src/util.ts

```diff
@@ -203,8 +202,19 @@ export const getInputJSON = (artifact: Artifact, options: Options, logger: Logge
     // If we're on Windows we need to de-Unixify the path so that Windows can read the file
     // We also need to replace the 'project:' prefix so that the file can be read
     const normalisedContractPath = normaliseContractPath(contractPath, options);
-    const absolutePath = require.resolve(normalisedContractPath, { paths: [options.projectDir] });
-    const content = fs.readFileSync(absolutePath, 'utf8');
+
+    let absolutePath;
+    try {
+      absolutePath = require.resolve(normalisedContractPath, { paths: [options.projectDir] });
+    } catch {
+      absolutePath = contractPath;
+    }
```

Comment:
Do you think it'd be possible to skip this whole "require.resolve" step and use `options.resolver.resolve(normalisedContractPath)` instead? Or do you think that there's some edge cases that would fail in that case?

Reply:
`resolver.resolve` needs an absolute path. This is something to change on Truffle's side, but will take some investigation and probably a breaking change to that package. I think for now we're stuck doing `require.resolve` here.

Reply:
This does probably warrant a comment, though:

```suggestion
    let absolutePath;
    try {
      absolutePath = require.resolve(normalisedContractPath, { paths: [options.projectDir] });
    } catch {
      // handle case where the contractPath refers to some ephemeral source
      // (e.g., truffle/console.sol)
      absolutePath = contractPath;
    }
```

Reply:
Yep that definitely helps!